server/resourcemanager/testutil/operations_list.go

```diff
@@ -1,109 +1,251 @@
+	queryString := buildQueryString(paginationParams)
 
-func (listSuccessOperation) assertResponse(t *testing.T, resp *http.Response, ct contentType, rt ResourceTypeTest) {
-	require.Equal(t, 200, resp.StatusCode)
+	url := fmt.Sprintf("%s/%s/%s", testServer.URL, strings.ToLower(resourceType), queryString)
 
-	jsonBody := responseBodyJSON(t, resp, ct)
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	require.NoError(t, err)
+	return req
+}
 
-	expected := `{
-		"count": 1,
-		"items": [` + ct.toJSON(rt.SampleJSON) + `]
-	}`
+const OperationListNoResults Operation = "ListNoResults"
 
-	require.JSONEq(t, expected, jsonBody)
+var listNoResultsOperation = operationTester{
+	name: OperationListNoResults,
+	buildRequest: func(t *testing.T, testServer *httptest.Server, ct contentTypeConverter, rt ResourceTypeTest) *http.Request {
+		return buildListRequest(
+			rt.ResourceType,
+			map[string]string{},
+			ct,
+			testServer,
+			t,
+		)
+	},
+	assertResponse: func(t *testing.T, resp *http.Response, ct contentTypeConverter, rt ResourceTypeTest) {
+		require.Equal(t, 200, resp.StatusCode)
+
+		jsonBody := responseBodyJSON(t, resp, ct)
+
+		expected := `{
+			"count": 0,
+			"items": []
+		}`
+
+		require.JSONEq(t, expected, jsonBody)
+	},
 }
 
-const OperationListInteralError Operation = "ListInteralError"
-
-type listInteralErrorOperation struct{}
+const OperationListSuccess Operation = "ListSuccess"
 
-func (op listInteralErrorOperation) postAssert(t *testing.T, ct contentType, rt ResourceTypeTest, testServer *httptest.Server) {
+var listSuccessOperation = operationTester{
+	name: OperationListSuccess,
+	buildRequest: func(t *testing.T, testServer *httptest.Server, ct contentTypeConverter, rt ResourceTypeTest) *http.Request {
+		return buildListRequest(
+			rt.ResourceType,
+			map[string]string{},
+			ct,
+			testServer,
+			t,
+		)
+	},
+	assertResponse: func(t *testing.T, resp *http.Response, ct contentTypeConverter, rt ResourceTypeTest) {
+		require.Equal(t, 200, resp.StatusCode)
+
+		jsonBody := responseBodyJSON(t, resp, ct)
+
+		expected := `{
+			"count": 1,
+			"items": [` + ct.toJSON(rt.SampleJSON) + `]
+		}`
+
+		require.JSONEq(t, expected, jsonBody)
+	},
 }
 
-func (op listInteralErrorOperation) buildRequest(t *testing.T, testServer *httptest.Server, ct contentType, rt ResourceTypeTest) *http.Request {
-	return buildListRequest(
-		rt.ResourceType,
-		ct,
-		testServer,
-		t,
-	)
+const OperationListPaginatedAscendingSuccess Operation = "ListPaginatedAscendingSuccess"
+
+var listPaginatedAscendingSuccessOperation = operationTester{
+	name: OperationListPaginatedAscendingSuccess,
+	buildRequest: func(t *testing.T, testServer *httptest.Server, ct contentTypeConverter, rt ResourceTypeTest) *http.Request {
+		return buildListRequest(
+			rt.ResourceType,
+			map[string]string{
+				"take":          "2",
+				"skip":          "1",
+				"sortBy":        getSortFieldsString(rt.PaginationSortFields),
```

Comment:
I think we should make one request per sorting field. I don't think we support sorting by multiple fields, so this request would actually fail

Reply:
Changed